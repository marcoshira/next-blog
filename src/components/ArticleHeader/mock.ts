import { PostMetadata } from '../../shared-types/postAttributes';

export default {
  id: '1',
  attributes: {
    title: 'Lorem ipsum dolor sit amet, consectetur adipiscing elit.',
    slug: 'teste-1',
    excerpt:
      'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Suspendisse auctor fermentum nisl, ut pharetra leo lacinia et. Etiam nec urna sed orci bibendum sodales.',
    allowComments: true,
    createdAt: '2022-03-25T18:18:34.944Z',
    updatedAt: '2022-03-27T20:41:54.493Z',
    tags: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'UseContext',
            slug: 'use-context',
          },
        },
        {
          id: '3',
          attributes: {
            displayName: 'UseState',
            slug: 'use-state',
          },
        },
      ],
    },
    categories: {
      data: [
        {
          id: '1',
          attributes: {
            displayName: 'JavaScript',
            slug: 'java-script',
          },
        },
        {
          id: '4',
          attributes: {
            displayName: 'React',
            slug: 'react',
          },
        },
      ],
    },
    author: {
      data: {
        id: '4',
        attributes: {
          displayName: 'Júlia Temari',
          slug: 'julia-temari',
        },
      },
    },
    cover: {
      data: {
        id: '14',
        attributes: {
          url: 'https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg',
          alternativeText: '1920x1080 1',
        },
      },
    },
    content:
      '<p>We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.</p><figure class="table"><table><thead><tr><th>Title 1</th><th>Title 2</th><th>Title 3</th><th>Title 4</th><th>Title 5</th></tr></thead><tbody><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr><tr><td>Content 1</td><td>Content 2</td><td>Content 3</td><td>Content 3</td><td>Content 3</td></tr></tbody></table></figure><blockquote><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</span></p></blockquote><p>&nbsp;</p><figure class="image"><img src="https://source.unsplash.com/random/800x800"></figure><p>&nbsp;</p><figure class="image image-style-side"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus. Imagem ao lado:&nbsp;</span>&nbsp;</p><p>&nbsp;</p><p>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);">Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla, </span>Teste de imagem centralizada</p><p>&nbsp;</p><figure class="image"><img src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg" alt="imagem 9"></figure><p><a href="https://google.com.br">Teste de link</a>&nbsp;</p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><i>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet. Fusce faucibus, ex quis facilisis congue, magna elit congue nulla,</i></span></p><p><span style="background-color:rgb(255,255,255);color:rgb(0,0,0);"><strong>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec sagittis nisl sagittis lectus ornare aliquet.</strong></span></p><p><code><strong>Teste de código asdasd asad</strong></code></p><pre><code class="language-plaintext">import { PostMetadata } from \'../../shared-types/postAttributes\';\nimport { ArticleHeader } from \'../ArticleHeader\';\nimport { HtmlContent } from \'../HtmlContent\';\nimport * as Styled from \'./styles\';\n\nexport type PostProps = PostMetadata;\n\nexport const Post = ({ attributes }: PostProps) =&gt; {\n  return (\n    &lt;Styled.Wrapper&gt;\n      &lt;ArticleHeader {...attributes} /&gt;\n      &lt;Styled.Content&gt;\n        &lt;HtmlContent html={attributes.content} /&gt;\n      &lt;/Styled.Content&gt;\n    &lt;/Styled.Wrapper&gt;\n  );\n};</code></pre><p><strong>Teste de vídeo</strong></p><figure class="media"><oembed url="https://www.youtube.com/watch?v=heRRKDQepy8"></oembed></figure>',
  },
} as PostMetadata;
