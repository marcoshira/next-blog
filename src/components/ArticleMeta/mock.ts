import { ArticleMetaProps } from '.';

export default {
  createdAt: '2022-03-25T18:28:16.052Z',
  updatedAt: '2022-03-28T18:29:04.677Z',
  tags: {
    data: [
      {
        id: '2',
        attributes: {
          displayName: 'UseRef',
          slug: 'use-ref',
        },
      },
      {
        id: '4',
        attributes: {
          displayName: 'Postgres',
          slug: 'postgres',
        },
      },
    ],
  },
  categories: {
    data: [
      {
        id: '5',
        attributes: {
          displayName: 'Django',
          slug: 'django',
        },
      },
    ],
  },
  author: {
    data: {
      id: '2',
      attributes: {
        displayName: 'Lu√≠sa Santiago',
        slug: 'luisa-santiago',
      },
    },
  },
} as ArticleMetaProps;
