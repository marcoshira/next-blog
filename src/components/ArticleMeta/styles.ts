import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.small};
    color: ${theme.colors.mediumGray};
    font-style: italic;
    a {
      color: ${theme.colors.crimson};
      text-decoration: none;
      transition: all 300ms ease-in-out;
      &:hover {
        filter: brightness(50%);
      }
    }
  `}
`;
