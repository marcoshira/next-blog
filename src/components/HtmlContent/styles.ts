import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    font-size: calc(${theme.font.sizes.small} + 0.2rem);
    line-height: 1.5;
    p {
      margin: ${theme.spacings.small} 0;
    }
    a,
    a:visited,
    a:link {
      color: ${theme.colors.crimson};
      text-decoration: none;
      transition: all 300ms ease-in-out;
    }
    a:hover {
      filter: brightness(50%);
    }
    code {
      font-family: monospace;
      color: ${theme.colors.crimson};
      font-size: ${theme.font.sizes.small};
      background: ${theme.colors.grey};
      padding: 0.2rem;
      margin: 0.2rem;
    }
    pre {
      background: ${theme.colors.black};
      padding: ${theme.spacings.medium};
      font-family: monospace;
      color: ${theme.colors.white};
      margin: ${theme.spacings.small} 0;
      width: 100%;
      overflow-x: auto;
      font-size: ${theme.font.sizes.small};
    }
    pre code {
      color: inherit;
      background: inherit;
    }
    img {
      max-width: 100%;
    }
    .image {
      line-height: 0;
      margin: ${theme.spacings.medium} 0;
    }
    .image figcaption {
      font-size: ${theme.font.sizes.small};
      padding: ${theme.spacings.small};
      text-align: center;
      line-height: 1.3;
    }
    .image-style-side {
      max-width: 50%;
      float: right;
      margin: ${theme.spacings.medium} 0;
    }
    hr {
      border: none;
      border-bottom: 0.1rem solid ${theme.colors.grey};
    }
    ul,
    ol {
      margin: ${theme.spacings.medium} ${theme.spacings.xlarge};
    }
    .table {
      width: 100%;
      overflow: hidden;
      overflow-x: auto;
    }
    table {
      width: 100%;
      border-collapse: collapse;
      margin: ${theme.spacings.small} 0;
    }
    table td,
    table th {
      padding: ${theme.spacings.small};
      border: 0.1rem solid ${theme.colors.grey};
    }
    blockquote {
      border-left: 0.5rem solid ${theme.colors.crimson};
      color: ${theme.colors.mediumGray};
      filter: brightness(80%);
      padding-left: ${theme.spacings.medium};
      font-style: italic;
      margin: ${theme.spacings.medium};
    }

    @media ${theme.media.lteMedium} {
      font-size: 2rem;
      .image-style-side {
        max-width: 100%;
        float: none;
        margin: 0;
      }
    }
  `}
`;
