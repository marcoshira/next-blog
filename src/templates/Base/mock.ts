import { BaseTemplateProps } from '.';
import { data } from '../../api/dados.json';

export default {
  settings: data.setting,
  children: 'Ol√°',
} as BaseTemplateProps;
