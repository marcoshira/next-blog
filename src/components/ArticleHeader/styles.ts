import styled, { css } from 'styled-components';
import { Wrapper as ArticleMeta } from '../ArticleMeta/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;

    > h1 {
      margin-bottom: ${theme.spacings.medium};
      max-width: 80rem;
    }

    > p {
      margin-top: 0;
      max-width: 80rem;
    }

    > img {
      width: 100%;
      max-width: 80rem;
    }

    & ${ArticleMeta} {
      max-width: 80rem;
    }
  `}
`;
