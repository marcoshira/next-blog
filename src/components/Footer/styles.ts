import styled, { css } from 'styled-components';
import { Container as HtmlContent } from '../HtmlContent/styles';

export const Container = styled.footer`
  ${({ theme }) => css`
    text-align: center;
    border-top: 0.2rem solid ${theme.colors.mediumGray};

    a {
      color: inherit;
      text-decoration: none;
    }

    & ${HtmlContent} {
      font-size: ${theme.font.sizes.small};

      > p > a {
        color: ${theme.colors.black};
      }
    }
  `}
`;
