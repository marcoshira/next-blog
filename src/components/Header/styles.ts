import styled, { css } from 'styled-components';
import { Title as Heading } from '../Heading/styles';

type DisplayText = {
  showText?: boolean;
};

const displayText = () => css`
  display: flex;
`;

export const Wrapper = styled.header`
  ${({ theme }) => css`
    display: flex;
    align-items: center;
    align-content: center;
    justify-content: center;
    width: 100%;

    div {
      margin: ${theme.spacings.xlarge} ${theme.spacings.medium};
    }
  `}
`;

export const Container = styled.div<DisplayText>`
  ${({ theme, showText }) => css`
    display: none;
    ${showText && displayText()}
    flex-direction: column;
    max-width: 70rem;

    & ${Heading} {
      margin-top: 0;
      margin-bottom: ${theme.spacings.xsmall};
    }
  `}
`;
