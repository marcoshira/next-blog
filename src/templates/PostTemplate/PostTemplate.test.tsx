import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { PostTemplate, PostTemplateProps } from '.';
import mock from './mock';

const props: PostTemplateProps = mock;

describe('<PostTemplate />', () => {
  it('should render', () => {
    const { container } = renderTheme(<PostTemplate {...props} />);

    expect(container).toMatchSnapshot();
  });
});
