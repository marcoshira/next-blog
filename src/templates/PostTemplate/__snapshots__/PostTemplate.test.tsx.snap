// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostTemplate /> should render 1`] = `
.c19 {
  font-size: 1.6rem;
  color: #AAAAAA;
  font-style: italic;
}

.c19 a {
  color: crimson;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c19 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c6 {
  color: inherit;
  font-size: 2.4rem;
  text-transform: uppercase;
}

.c12 {
  color: inherit;
  font-size: 2.4rem;
  text-transform: none;
}

.c17 {
  color: inherit;
  font-size: 4.0rem;
  text-transform: none;
}

.c16 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-bottom: 0.1rem solid #DDDDDD;
  padding-bottom: 4.0rem;
  margin-bottom: 4.0rem;
}

.c16 > h1 {
  margin-bottom: 2.4rem;
  max-width: 80rem;
}

.c16 > p {
  margin-top: 0;
  max-width: 80rem;
  font-size: calc(1.6rem + 0.4rem);
}

.c16 > img {
  width: 100%;
  max-width: 80rem;
}

.c16 .c18 {
  max-width: 80rem;
}

.c16 .c18 > p {
  margin-bottom: 0;
}

.c23 {
  font-size: calc(1.6rem + 0.2rem);
  line-height: 1.5;
}

.c23 p {
  margin: 1.6rem 0;
}

.c23 a,
.c23 a:visited,
.c23 a:link {
  color: crimson;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c23 a:hover {
  -webkit-filter: brightness(50%);
  filter: brightness(50%);
}

.c23 code {
  font-family: monospace;
  color: crimson;
  font-size: 1.6rem;
  background: #DDDDDD;
  padding: 0.2rem;
  margin: 0.2rem;
}

.c23 pre {
  background: #000000;
  padding: 2.4rem;
  font-family: monospace;
  color: #FFFFFF;
  margin: 1.6rem 0;
  width: 100%;
  overflow-x: auto;
  font-size: 1.6rem;
}

.c23 pre code {
  color: inherit;
  background: inherit;
}

.c23 img {
  max-width: 100%;
}

.c23 .image {
  line-height: 0;
  margin: 2.4rem 0;
}

.c23 .image figcaption {
  font-size: 1.6rem;
  padding: 1.6rem;
  text-align: center;
  line-height: 1.3;
}

.c23 .image-style-side {
  max-width: 50%;
  float: right;
  margin: 2.4rem 0;
}

.c23 hr {
  border: none;
  border-bottom: 0.1rem solid #DDDDDD;
}

.c23 ul,
.c23 ol {
  margin: 2.4rem 4.0rem;
}

.c23 .table {
  width: 100%;
  overflow: hidden;
  overflow-x: auto;
}

.c23 table {
  width: 100%;
  border-collapse: collapse;
  margin: 1.6rem 0;
}

.c23 table td,
.c23 table th {
  padding: 1.6rem;
  border: 0.1rem solid #DDDDDD;
}

.c23 blockquote {
  border-left: 0.5rem solid crimson;
  color: #AAAAAA;
  -webkit-filter: brightness(80%);
  filter: brightness(80%);
  padding-left: 2.4rem;
  font-style: italic;
  margin: 2.4rem;
}

.c15 {
  width: 100%;
  margin: 0 auto;
  padding: 0 3.2rem;
  max-width: 96rem;
}

.c20 {
  width: 100%;
  margin: 0 auto;
  padding: 0 3.2rem;
  max-width: 80rem;
}

.c14 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
}

.c21 .c22 {
  max-width: 80rem;
  font-size: 1.6rem;
}

.c25 {
  text-align: center;
  border-top: 0.2rem solid #AAAAAA;
}

.c25 a {
  color: inherit;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c25 .c22 {
  font-size: 1.6rem;
}

.c25 .c22 > p > a {
  color: #000000;
}

.c26 {
  background: #FFFFFF;
  color: #0A1128;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 4rem;
  height: 4rem;
  position: fixed;
  bottom: 2rem;
  right: 2rem;
  z-index: 6;
  opacity: 0.7;
  border-radius: 2rem;
  border: 3px solid #0A1128;
}

.c1 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
}

.c7 > img {
  height: 12rem;
  width: 12rem;
  border-radius: 50%;
}

.c10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-align-content: center;
  -ms-flex-line-pack: center;
  align-content: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  width: 100%;
}

.c10 div {
  margin: 4.0rem 2.4rem;
}

.c11 {
  display: none;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 70rem;
}

.c11 .c5 {
  margin-top: 0;
  margin-bottom: 0.8rem;
}

.c2 {
  background: #000000;
  padding: 3.2rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: fixed;
  z-index: 1;
  width: 100%;
  max-width: 32rem;
  height: 100vh;
  top: 0;
  left: 0;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  overflow-y: auto;
  left: -32rem;
  overflow-y: hidden;
}

.c3 {
  margin: auto;
  width: 100%;
}

.c4 .c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0;
  margin-bottom: 4.8rem;
}

.c4 .c5 img {
  border: 0.3rem solid crimson;
}

.c0 {
  position: fixed;
  top: 2.4rem;
  color: #FFFFFF;
  background: #000000;
  z-index: 2;
  width: 3rem;
  height: 3rem;
  left: 26rem;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
  left: 2rem;
  color: #000000;
  background: inherit;
}

.c8 {
  display: block;
  color: #FFFFFF;
  -webkit-text-decoration: none;
  text-decoration: none;
  margin-bottom: 1.6rem;
  font-size: 1.8rem;
  border-right: 0.5rem solid #000000;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c8:hover {
  border-right: 0.5rem solid crimson;
  color: crimson;
}

.c9 {
  margin-top: 6.4rem;
  padding: 0 3.2rem;
}

.c13 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  margin-bottom: 6.4rem;
}

.c24 {
  max-width: 120rem;
  width: 100%;
  margin: 0 auto;
  padding: 3.2rem;
}

@media (max-width:768px) {
  .c23 {
    font-size: 2rem;
  }

  .c23 .image-style-side {
    max-width: 100%;
    float: none;
    margin: 0;
  }
}

@media (max-width:768px) {
  .c15 {
    padding: 0 calc(3.2rem / 2);
  }
}

@media (max-width:768px) {
  .c20 {
    padding: 0 calc(3.2rem / 2);
  }
}

<div>
  <div
    class=""
  >
    <a
      aria-label="Open or close menu"
      class="c0"
      href="#"
      title="Open or close menu"
    >
      <svg
        aria-hidden="true"
        aria-label="Open menu"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
        />
      </svg>
    </a>
    <div
      aria-hidden="true"
      class="c2"
    >
      <nav
        class="c3"
      >
        <div
          class="c4"
        >
          <h1
            class="c5 c6"
          >
            <a
              class="c7"
              href="/"
              target="_self"
            >
              <img
                alt="Blog Project"
                src="https://source.unsplash.com/random/800x800"
              />
            </a>
          </h1>
        </div>
        <a
          class="c8"
          href="https://marcoshirazawa.netlify.app/"
          target="_blank"
        >
          Contato
        </a>
        <a
          class="c8"
          href="https://landing-page-next-mmh.vercel.app/#home"
          target="_blank"
        >
          Último projeto
        </a>
      </nav>
    </div>
    <div
      class="c9"
    >
      <header
        class="c10"
      >
        <h1
          class="c5 c6"
        >
          <a
            class="c7"
            href="/"
            target="_self"
          >
            <img
              alt="Blog Project"
              src="https://source.unsplash.com/random/800x800"
            />
          </a>
        </h1>
        <div
          class="c11"
        >
          <h1
            class="c5 c12"
          >
            Blog Project
          </h1>
          Descrição do blog.
        </div>
      </header>
    </div>
    <div
      class="c13"
    >
      <article
        class="c14"
      >
        <div
          class="c15"
        >
          <div
            class="c16"
          >
            <h1
              class="c5 c17"
            >
              Teste 1
            </h1>
            <p>
              Teste de post.
            </p>
            <img
              alt="1920x1080 1"
              src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1648156169/1920x1080_6aaecb929e.jpg"
            />
            <div
              class="c18 c19"
            >
              <p>
                <span>
                  Por 
                </span>
                <a
                  href="/author/julia-temari"
                >
                  Júlia Temari
                </a>
                <span
                  class="separator"
                >
                   | 
                </span>
                <time
                  datetime="2022-03-25T18:18:34.944Z"
                >
                  25/03/2022 15:18
                </time>
                <span
                  class="separator"
                >
                   | 
                </span>
                <span>
                  Editado em:
                   
                  <time
                    datetime="2022-03-27T20:41:54.493Z"
                  >
                    27/03/2022 17:41
                  </time>
                </span>
                <span
                  class="separator"
                >
                   | 
                </span>
                <span
                  class="categories"
                >
                  <span>
                    Categorias: 
                  </span>
                  <span>
                    <a
                      href="/category/java-script"
                    >
                      JavaScript
                    </a>
                    <span>
                      , 
                    </span>
                  </span>
                  <span>
                    <a
                      href="/category/react"
                    >
                      React
                    </a>
                  </span>
                </span>
                <span
                  class="separator"
                >
                   | 
                </span>
                <span
                  class="tags"
                >
                  <span>
                    Tags: 
                  </span>
                  <span>
                    <a
                      href="/tag/use-context"
                    >
                      UseContext
                    </a>
                    <span>
                      , 
                    </span>
                  </span>
                  <span>
                    <a
                      href="/tag/use-state"
                    >
                      UseState
                    </a>
                  </span>
                </span>
              </p>
            </div>
          </div>
        </div>
        <div
          class="c20"
        >
          <div
            class="c21"
          >
            <div
              class="c22 c23"
            >
              <p>
                We’ll start with some background on the Mac transition and give an overview of Firefox internals that needed to know about the new architecture, before moving on to the concept of Universal Binaries.
              </p>
              <figure
                class="table"
              >
                <table>
                  <thead>
                    <tr>
                      <th>
                        Title 1
                      </th>
                      <th>
                        Title 2
                      </th>
                      <th>
                        Title 3
                      </th>
                      <th>
                        Title 4
                      </th>
                      <th>
                        Title 5
                      </th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <td>
                        Content 1
                      </td>
                      <td>
                        Content 2
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Content 1
                      </td>
                      <td>
                        Content 2
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                    </tr>
                    <tr>
                      <td>
                        Content 1
                      </td>
                      <td>
                        Content 2
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                      <td>
                        Content 3
                      </td>
                    </tr>
                  </tbody>
                </table>
              </figure>
              <blockquote>
                <p>
                  Teste de citação
                </p>
              </blockquote>
              <p>
                 
              </p>
              <p>
                Teste de foto ao lado
              </p>
              <figure
                class="image"
              >
                <img
                  alt="imagem 11"
                  src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892307/800x800_74f76986f2.jpg"
                />
              </figure>
              <p>
                 
              </p>
              <p>
                Teste de imagem centralizada
              </p>
              <p>
                 
              </p>
              <p>
                 
              </p>
              <figure
                class="image"
              >
                <img
                  alt="imagem 9"
                  src="https://res.cloudinary.com/dr5ot2zqr/image/upload/v1647892277/800x800_a1b3864dfd.jpg"
                />
              </figure>
              <p>
                <a
                  href="https://google.com.br"
                >
                  Teste de link
                </a>
              </p>
              <p>
                 
              </p>
              <p>
                <i>
                  Itálico
                </i>
              </p>
              <p>
                 
              </p>
              <p>
                <strong>
                  Negrito
                </strong>
              </p>
              <p>
                 
              </p>
              <p>
                <code>
                  <strong>
                    Teste de código
                  </strong>
                </code>
              </p>
              <p>
                 
              </p>
              <pre>
                <code
                  class="language-plaintext"
                >
                  Teste de bloco de código
                </code>
              </pre>
              <p>
                 
              </p>
              <p>
                <strong>
                  Teste de vídeo
                </strong>
              </p>
              <p>
                 
              </p>
              <iframe
                height="315"
                src="https://www.youtube.com/embed/qF1DTK4U1AM"
                width="560"
              >
                &lt;/oembed&gt;&lt;/figure" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen&gt;
              </iframe>
            </div>
          </div>
        </div>
      </article>
    </div>
    <div
      class="c24"
    >
      <footer
        class="c25"
      >
        <div
          class="c22 c23"
        >
          <p>
            <a
              href="https://marcoshirazawa.netlify.app/"
            >
              <i>
                Feito com ❤ por Marcos Hirazawa
              </i>
            </a>
          </p>
        </div>
      </footer>
    </div>
    <a
      aria-label="Go to top"
      class="c26"
      href="#"
      title="Go to top"
    >
      <svg
        aria-hidden="true"
        class="c1"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0V0z"
          fill="none"
        />
        <path
          d="M7.41 15.41L12 10.83l4.59 4.58L18 14l-6-6-6 6 1.41 1.41z"
        />
      </svg>
    </a>
  </div>
</div>
`;
